### PR TITLE
DependencyRule: Improve level-related documentation

### PR DESCRIPTION
**Pull request by sschuberth**
Avoid the term "root", as the root of a dependency tree usually is the
project itself. However, here "level 0" refers to the direct
dependencies of a project.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

**Comment by sschuberth** [comment]
@mmurto, does this make it clear enough?

File: evaluator/src/main/kotlin/DependencyRule.kt [hunk 1]
```diff
@@ -45,13 +45,13 @@ class DependencyRule(
     val dependency: PackageReference,
 
     /**
-     * The ancestors of the [dependency] in the dependency tree. The first entry is the root of the tree, the last entry
-     * (at [level] - 1) is the direct parent.
+     * The ancestors of the [dependency] in the dependency tree, sorted from farthest to closest: The first entry is the
+     * direct dependency of a project, the last entry (at index [level] - 1) is the direct parent of this [dependency].
```
Comment (fviernau) [inline evaluator/src/main/kotlin/DependencyRule.kt:49]: How about removing "(at index [level] - 1)" as "the last entry" already describes it sufficiently?

**Review by fviernau — APPROVED** [review]

File: evaluator/src/main/kotlin/DependencyRule.kt (hunk 1 above)
Comment (sschuberth) [reply evaluator/src/main/kotlin/DependencyRule.kt:49]: I liked it, as it emphasizes that the number of entries in `ancestors` equals the `level`.

**Comment by mmurto** [comment]
I think this should also be corrected to `isAtTreeLevel(0)`.

https://github.com/oss-review-toolkit/ort/blob/fd32da954a63ae7ceaf0e1b8d81af7e8718682e5/examples/rules.kts#L230-L234

**Comment by mnonnenmacher** [comment]
> I think this should also be corrected to `isAtTreeLevel(0)`.
> 
> https://github.com/oss-review-toolkit/ort/blob/fd32da954a63ae7ceaf0e1b8d81af7e8718682e5/examples/rules.kts#L230-L234

@sschuberth Could you add this fix to your PR?

File: evaluator/src/main/kotlin/DependencyRule.kt (hunk 1 above)
Comment (mnonnenmacher) [reply evaluator/src/main/kotlin/DependencyRule.kt:49]: Maybe write it as "(at index `level - 1`)".

Comment (mnonnenmacher) [reply evaluator/src/main/kotlin/DependencyRule.kt:49]: You could make it even clearer by adding: "If the list is empty it means that this dependency is a direct dependency."

File: evaluator/src/main/kotlin/DependencyRule.kt [hunk 2]
```diff
@@ -45,13 +45,13 @@ class DependencyRule(
...
      */
     val ancestors: List<PackageReference>,
 
     /**
-     * The level of the [dependency] inside the dependency tree. Starts with 0 for root level entries.
+     * The level of the [dependency] inside the dependency tree. Starts with 0 for the direct dependency of a project.
```
Comment (mnonnenmacher) [inline evaluator/src/main/kotlin/DependencyRule.kt:54]: Replace "the" with "a", otherwise it sounds like there can only be one direct dependency.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]
See above...

**Review by mnonnenmacher — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite the unrelated test failure.

**Pull request merged**